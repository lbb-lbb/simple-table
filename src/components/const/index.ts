/** @format */

export const SORT_ITEM = {
    normal: 'sort-normal',
    asc: 'sort-asc',
    desc: 'sort-desc',
}
