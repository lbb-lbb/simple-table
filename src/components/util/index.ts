/** @format */

export function addHeaderSlotName(value: string) {
    return `header-${value}`
}
