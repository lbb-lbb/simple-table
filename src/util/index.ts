
export function addHeaderSlotName(value: string) {
    return `header-${value}`
}
/** 存在小数位数返回true */
export function isInteger(n: string){
    return parseInt(n) === parseFloat(n)
}

export function info(message: unknown) {
    if (import.meta.env.MODE !== 'production') {
        console.info(message + '\n----------------' + loggerMessage())
    }
}

export function err(message: unknown) {
    if (import.meta.env.MODE !== 'production') {
        console.error(message + '\n----------------' + loggerMessage())
    }
}

export function warn(message: unknown) {
    if (import.meta.env.MODE !== 'production') {
        console.warn(message + '\n----------------' + loggerMessage())
    }
}
function loggerMessage() {
    return `${new Date().toLocaleDateString()} ${new Date().toLocaleTimeString()}`
}


