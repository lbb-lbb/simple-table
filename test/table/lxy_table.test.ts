import { expect, test } from "vitest";
import lxyTable from "../../src/components/lxy_table/lxy_table.vue";
import {myAllWrapper, myWrapper} from "../util";
import {COLUMNS, DATA} from "../const";

test("mount component", async () => {
  const wrapper = myAllWrapper(lxyTable, {
    props: {
      columns: COLUMNS,
      data: DATA
    },
  });

  // 点击排序图标
  await wrapper.findAll('.header-filter')[0].trigger('click')

  expect(wrapper.html()).toMatchSnapshot();
});

test("使用name列插槽以及操作列插槽", () => {
  const wrapper = myAllWrapper(lxyTable, {
    props: {
      columns: COLUMNS,
      data: DATA,
      openOption: true
    },
    slots: {
      name: '<div class="name">name列内容</div>',
      'header-name': '<div class="header-name">name列表头</div>',
      options: '<div class="body-options">表格操作列内容</div>',
      'header-options': '<div class="header-options">表头操作列内容</div>'
    }
  });
  expect(wrapper.find('.header-options').exists()).toBe(true)
  expect(wrapper.find('.body-options').exists()).toBe(true)
  expect(wrapper.find('.header-name').exists()).toBe(true)
  expect(wrapper.find('.name').exists()).toBe(true)
  expect(wrapper.html()).toMatchSnapshot();
});

test("传入空表头", async () => {
  function html() {
    myWrapper(lxyTable)
  }

  expect(html).toThrowError('传入的columns必须是个非空数组');
})

test('options插槽名冲突', () => {
  function html() {
    myWrapper(lxyTable, {
      props: {
        columns: [{value: 'options', label: 'Name'}],
        openOption: true
      }
    })
  }

  expect(html).toThrowError('开启的操作列插槽名"options"与columns里的value存在重名');
})

test('empty插槽名冲突', () => {
  function html() {
    myWrapper(lxyTable, {
      props: {
        columns: [{value: 'empty', label: 'Name'}]
      }
    })
  }

  expect(html).toThrowError('为空的插槽名"empty"与columns里的value存在重名');
})


