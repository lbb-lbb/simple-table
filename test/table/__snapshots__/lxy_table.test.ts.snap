// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div>
  <table class=\\"table\\">
    <thead data-v-5f17bdb8=\\"\\">
      <tr data-v-5f17bdb8=\\"\\">
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">Name<span class=\\"header-filter sort-asc\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">Age<span class=\\"header-filter\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">Manager<span class=\\"header-filter\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">Start Date<span class=\\"header-filter\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">时间<span class=\\"header-filter\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <!--v-if-->
      </tr>
    </thead>
    <tbody class=\\"table-body\\">
      <tr>
        <td>Frank Rossitano</td>
        <td>36</td>
        <td>false</td>
        <td></td>
        <td>Thu May 14 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <!--v-if-->
      </tr>
      <tr>
        <td>Jenna Maroney</td>
        <td>40</td>
        <td>false</td>
        <td>02-28-1999</td>
        <td>Thu May 07 1998 00:00:00 GMT+0800 (中国标准时间)</td>
        <!--v-if-->
      </tr>
      <tr>
        <td>Kenneth Parcell</td>
        <td>Infinity</td>
        <td>false</td>
        <td></td>
        <td>Sat May 30 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <!--v-if-->
      </tr>
      <tr>
        <td>Liz Lemon</td>
        <td>36</td>
        <td>true</td>
        <td>02-28-1999</td>
        <td>Sat May 16 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <!--v-if-->
      </tr>
      <tr>
        <td></td>
        <td>12</td>
        <td></td>
        <td></td>
        <td>Mon May 11 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <!--v-if-->
      </tr>
      <tr>
        <td>Tracy Morgan</td>
        <td>39</td>
        <td>false</td>
        <td>07-12-2002</td>
        <td>Wed May 06 1998 00:00:00 GMT+0800 (中国标准时间)</td>
        <!--v-if-->
      </tr>
      <tr>
        <td>啊啊</td>
        <td></td>
        <td>true</td>
        <td>04-01-2000</td>
        <td>Thu May 28 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <!--v-if-->
      </tr>
      <tr>
        <td>阀分撒</td>
        <td>40</td>
        <td>true</td>
        <td>03-05-1997</td>
        <td>Sun May 17 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <!--v-if-->
      </tr>
      <!--v-if-->
    </tbody>
  </table>
</div>"
`;

exports[`不传入表头 1`] = `
"<div>
  <table class=\\"table\\">
    <table-header-stub openoption=\\"false\\"></table-header-stub>
    <table-body-stub data=\\"\\" orderby=\\"\\" order=\\"sort-normal\\" openoption=\\"false\\"></table-body-stub>
  </table>
</div>"
`;

exports[`使用name列插槽以及操作列插槽 1`] = `
"<div>
  <table class=\\"table\\">
    <thead data-v-5f17bdb8=\\"\\">
      <tr data-v-5f17bdb8=\\"\\">
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">
            <div class=\\"header-name\\">name列表头</div><span class=\\"header-filter sort-normal\\" data-v-5f17bdb8=\\"\\"></span>
          </div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">Age<span class=\\"header-filter\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">Manager<span class=\\"header-filter\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">Start Date<span class=\\"header-filter\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div style=\\"display: flex;\\" class=\\"header-wrapper\\" data-v-5f17bdb8=\\"\\">时间<span class=\\"header-filter\\" data-v-5f17bdb8=\\"\\"></span></div>
        </th>
        <th data-v-5f17bdb8=\\"\\">
          <div class=\\"header-options\\">表头操作列内容</div>
        </th>
      </tr>
    </thead>
    <tbody class=\\"table-body\\">
      <tr>
        <td>
          <div class=\\"name\\">name列内容</div>
        </td>
        <td>36</td>
        <td>true</td>
        <td>02-28-1999</td>
        <td>Sat May 16 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <td>
          <div class=\\"body-option\\">
            <div class=\\"body-options\\">表格操作列内容</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div class=\\"name\\">name列内容</div>
        </td>
        <td>40</td>
        <td>true</td>
        <td>03-05-1997</td>
        <td>Sun May 17 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <td>
          <div class=\\"body-option\\">
            <div class=\\"body-options\\">表格操作列内容</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div class=\\"name\\">name列内容</div>
        </td>
        <td>39</td>
        <td>false</td>
        <td>07-12-2002</td>
        <td>Wed May 06 1998 00:00:00 GMT+0800 (中国标准时间)</td>
        <td>
          <div class=\\"body-option\\">
            <div class=\\"body-options\\">表格操作列内容</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div class=\\"name\\">name列内容</div>
        </td>
        <td>40</td>
        <td>false</td>
        <td>02-28-1999</td>
        <td>Thu May 07 1998 00:00:00 GMT+0800 (中国标准时间)</td>
        <td>
          <div class=\\"body-option\\">
            <div class=\\"body-options\\">表格操作列内容</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div class=\\"name\\">name列内容</div>
        </td>
        <td>Infinity</td>
        <td>false</td>
        <td></td>
        <td>Sat May 30 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <td>
          <div class=\\"body-option\\">
            <div class=\\"body-options\\">表格操作列内容</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div class=\\"name\\">name列内容</div>
        </td>
        <td></td>
        <td>true</td>
        <td>04-01-2000</td>
        <td>Thu May 28 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <td>
          <div class=\\"body-option\\">
            <div class=\\"body-options\\">表格操作列内容</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div class=\\"name\\">name列内容</div>
        </td>
        <td>36</td>
        <td>false</td>
        <td></td>
        <td>Thu May 14 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <td>
          <div class=\\"body-option\\">
            <div class=\\"body-options\\">表格操作列内容</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div class=\\"name\\">name列内容</div>
        </td>
        <td>12</td>
        <td></td>
        <td></td>
        <td>Mon May 11 1998 08:00:00 GMT+0800 (中国标准时间)</td>
        <td>
          <div class=\\"body-option\\">
            <div class=\\"body-options\\">表格操作列内容</div>
          </div>
        </td>
      </tr>
      <!--v-if-->
    </tbody>
  </table>
</div>"
`;
